src/modules/file-upload-input.marko

```diff
@@ -0,0 +1,111 @@
+<div id="file-upload-input">
+    <section-header metadata=input.metadata name=input.name alpha="true"/>
+
+    <p>
+        The file-upload-input allows users to upload one or more files to a location. To make the entire component clickable and open the browse file dialog, a label is used to wrap the display elements and the input is hidden. The input element id should be${" "}
+        <span class="highlight">
+            file-upload-input__input
+        </span>
+         and the label for attribute should match the input id.
+    </p>
+    <p>
+        Apply the
+        <span class="highlight">
+            file-upload-input__upload-icon
+        </span>
+         class to set the upload icon.
+    </p>
+    <p>
+        Apply the
+        <span class="highlight">
+            file-upload-input__content
+        </span>
+        ${" "}class to set the content for the file-upload-input to provide upload instructions. This can include the header, subheader and a CTA button. If a button is defined, it should trigger the browse file dialog.
+    </p>
+    <p>
+        Apply the
+        <span class="highlight">
+            file-upload-input__content-header
+        </span>
+         class to set the content header.
+    </p>
+    <p>
+        Apply the
+        <span class="highlight">
+            file-upload-input__content-subheader
+        </span>
+         class to set the content subheader.
+    </p>
+    <p>
+        The component will change to a compact version for small screens at the medium breakpoint.
+    </p>
+
+    <h4>Drag and Drop</h4>
+    <p>
+        Since a label element is used to achieve the UI aesthetic and for a11y support, the drag and drop interaction will need to be added with JS on the label element.${" "}
+    </p>
+
+    <h3>Base</h3>
+
+    <div class="demo">
+        <div class="demo__inner">
+            <div class="file-upload-input" tabindex="0">
```

Comment:
If we put a tab index on an element, that element is going to need a role for assistive technology. I don't think there is a good corresponding role for what we are trying to achieve here. We may need to revisit this aspect. I was anticipating that the keyboard focus would only go on the input type=file, but I see it landing on this container and a nested button.

Reply:
I was following this solution for styling the input type=file

https://stackoverflow.com/questions/572768/styling-an-input-type-file-button

The container is a `<label>` which is for the input type=file element, doesn't that help describe the input type=file? I agree we can possibly clip it instead of making it hidden